src/Components/WebAssembly/WebAssembly.Authentication/src/RemoteAuthenticatorViewCore.cs

```diff
@@ -282,9 +278,7 @@ private async Task ProcessLogInCallback()
 
     private async Task ProcessLogOut(string returnUrl)
     {
-        if ((Navigation.HistoryEntryState != null && !ValidateSignOutRequestState()) ||
-            // For backcompat purposes, keep SignOutManager working, even though we now use the history.state for this.
-            (Navigation.HistoryEntryState == null && !await SignOutManager.ValidateSignOutState()))
+        if (Navigation.HistoryEntryState != null && !ValidateSignOutRequestState())
```

Comment:
Should this be `Navigation.HistoryEntryState == null || !ValidateSignOutRequestState()` instead?